Src/FluentAssertions/Types/TypeAssertions.cs

```diff
@@ -462,18 +462,7 @@ public AndConstraint<TypeAssertions> Implement(Type interfaceType, string becaus
     {
         Guard.ThrowIfArgumentIsNull(interfaceType);
 
-        bool containsInterface = interfaceType.IsAssignableFrom(Subject) && interfaceType != Subject;
-
-        Execute.Assertion
-            .BecauseOf(because, becauseArgs)
-            .WithExpectation("Expected type {0} to implement interface {1}{reason}", Subject, interfaceType)
-            .ForCondition(interfaceType.IsInterface)
-            .FailWith(", but {0} is not an interface.", interfaceType)
-            .Then
-            .ForCondition(containsInterface)
-            .FailWith(", but it does not.")
-            .Then
-            .ClearExpectation();
+        ImplementInternal(interfaceType, because, becauseArgs);
```

Comment:
This is just a suggestion.. could most definitely be a better name...

`SubjectImplenents(...)` ?
`DoesImplement(...)` ?

Reply:
I have called it `SubjectImplenents()`